### ISSUE TITLE
Inconsistent Relay connection resolutions

### ISSUE THREAD

**Issue by antoniobg** [rmosolgo/graphql-ruby#419]
We've found this problem where calling `resolve` in a connection return different things depending if we execute it in local (development or test env) or if we execute in production or Codeship. 

```ruby
GraphQL::Forum::FetchForumCategoriesField = GraphQL::Field.define do
  description 'Fetches the categories'
  type GraphQL::ForumCategoryType.connection_type

  resolve -> (_obj, _args, _ctx) {
    Forum::Categories.all
  }
end
``` 

Calling `resolve` would return an `ActiveRecord::Relation` in local, but a `GraphQL::Relay::RelationConnection`. Does anybody have a hint about what could be the reason?

**Comment by rmosolgo** [comment]
Hi, sorry about that! I _think_ this is fixed in 1.2.5 (after https://github.com/rmosolgo/graphql-ruby/pull/399/commits/266f2dfa7099098db6703ce8bb8f4a9bac8be282)

The reason is because when you use the connection helper, eg 

```ruby 
connection :categories, FetchForumCategoriesField
```

It would "reach in" to the field and modify the `resolve` function. So, if you called `resolve` _before_ `connection` was called, you'd get a `Relation`. But if you called `resolve` _after_ `connection`, you'd get a `RelationConnection`. 

What a mess! Are you able to give it a try on 1.2.5?

**Comment by antoniobg** [comment]
Thank you @rmosolgo, that worked 👍
